man/ola_plugin_info.1
.TH ola_plugin_info 1 "June 2015"
.SH NAME
ola_plugin_info \- Get info on the plugins loaded by olad.
.SH SYNOPSIS
.B ola_plugin_info
[--plugin-id <plugin-id>] [--list-plugin-ids]
.SH DESCRIPTION
.B ola_plugin_info
is used to get info on the plugins loaded by olad. With no arguments it shows
the name and id of the loaded plugins.
.SH OPTIONS
.IP "-h, --help"
Display the help message
.IP "-p, --plugin-id <plugin_id>"
Id of the plugin to fetch the description of.
.IP "--list-plugin-ids"
List just the ids of the plugins loaded by olad.
.SH SEE ALSO
.BR olad(1) ,
.


NEWS
x/y/2023 ola-0.10.9
 Features:
 * Python 3 compatibility across the board (including the RDM Responder Tests)!
   #1506
 * Support for the JMS USB2DMX PRO V2.1 device via the FTDI plugin #1728

 API:
 * Python: Add a fetch current DMX example.

 RDM Tests:
 * Python 3 compatibility of the RDM Tests #1599
 * Fix a longstanding bug in the GetMaxPacketSize RDM test around timeouts

 Bugs:
 * Fix some tests not working on Big Endian machines because our RPC is encoded
   with native format and those tests used sample data from a Little Endian
   machine
 * Renamed EndpointNoticationEvent(sic) to EndpointNotificationEvent in the
   E1.33 EndpointManager code
 * Check if librt exists and add it to the build if required
 * Ensure we can build on later Protobuf and C++ and silence deprecation
   messages where required
 * Fix the NetBSD build
 * Clarify the usage of some of the various Enttec Open DMX USB plugins
 * Relax the timing on the ClientWrapperTest so it doesn't fail occasionally
   #1714
 * Correct the documentation of some existing limits in the E1.31 plugin
 * Fix some undefined behaviour in TimeoutManager::ExecuteTimeouts #1726
 * Fix UARTs on Debian Bullseye #1749
 * Fix Python version detection for versions > 3.9 #1755, #1780
 * Update Angular and Angular-Route versions used by the new web UI and add
   workaround to be compatible #1782 Debian #1016447
 * Significantly reduce the delay when using StreamingClient with EPoll and
   multiple universes #1793

 Internal:
 * Add more debugging so we can tell why setting the baud rate failed
 * Replace "currentThread" with "current_thread" in Python since the former has
   been deprecated
 * Fix lots more internal typos and grammar
 * Add unit tests for more of the Python RDM Test classes
 * Sort the Python imports with isort

22/11/2020 ola-0.10.8
 Features:
 * Support DragonFly BSD
 * Add another Debian test
 * Add the E1.33 NACK Reasons
 * Add the E1.37-7 NACK Reasons
 * Use clock_gettime's CLOCK_MONOTONIC if available
 * Collect the PDL size in the model collector

 API:
 * Deprecated Clock::CurrentTime and added Clock::CurrentMonotonicTime and 
   Clock::CurrentRealTime. Occurrences of Clock::CurrentTime have been
   replaced with either variant as applicable.

 RDM Tests:
 * Fix a bug in the RDM tests to ensure they can cope with 255 personalities
   #1508
 * Only add valid network interface IDs to the test list so future tests don't
   fail
 * Add a label to the GetOutOfRangePersonalityDescription test

 Bugs:
 * Fix a few more minor documentation spelling errors found by codespell
 * Point to the main site rather than the wiki for RDM test help
 * Fix the build on FreeBSD 10.0, 10.4 and 11
 * Make PID store loading more robust
 * Clarify some of the man page options for olad
 * Fix some dead links in documentation
 * Properly close socket on destruction of Python OlaClient
 * Fix various Python 3 compatibility issues
 * Try and fetch index and type via sockaddr_dl struct for better *BSD support
 * Backport some Debian changes
 * Use C++11 if required by random #1477
 * Correct uartdmx plugin documentation for the example UART #1320
 * Ensure we don't return unsuitable network indices via RDM #1478
 * Ensure a non-existent UID exits with a non-zero exit code
 * Fix a bug in how we simulate RDM responses and collisions
 * Fix an RDM discovery bug for how we handle phantom UIDs
 * Ensure _close_callback is assigned before we might need to run it in the
   Python API
 * Fix a bug with the Python ola_rdm_get and PROXIED_DEVICES display
 * Renamed RESONSE_INVALID_DESTINATION(sic) to RESPONSE_INVALID_DESTINATION in
   the ArduinoWidget code
 * Fix compatibility with GCC 9 #1553 Debian #925793
 * Ensure the GPIO plugin correctly writes to the last configured pin
 * Fix compatibility with ncurses 6
 * Fix compatibility with Protobuf 3.7 and newer (tested with up to 3.12.2)
   #1550, #1634
 * Rename CircularDepdendancyException(sic) to CircularDependencyException in
   the Python RDM Test code, also the relevant comments
 * Fix a minor longstanding logging bug with Avahi subtype registration errors
 * Fix Enttec USB Pro Mk2 firmware >=4.15 RDM Discovery #1631
 * Fix compatibility with libmicrohttpd v0.9.71 and newer #1650
 * Fix some Doxygen escaping
 * Rename AppendMultipler(sic) to AppendMultiplier in the RDM messaging code
 * Fix a few edge cases in ola_trigger around whitespace in the config files
 * Fix compatibility with clang 7 and newer (tested with up to 11) #1564

 Internal:
 * Replace "readdir_r" with "readdir" since the former has been deprecated
 * Add a unit test for the functions that used readdir_r before
 * Add a new 'make flake8' target #1619
 * Add a new 'make cpplint' target
 * Add a new 'make lint' target which runs the flake8 and cpplint runs

13/7/2018 ola-0.10.7
 Features:
 * Allow multiple KiNet ports to be patched to the same universe #1414
 * Allow multiple FTDI ports to be patched to the same universe
 * Support libftdi1 as well as libftdi0 in FTDI DMX #1012 Debian #810374
 * Support FTDI DMX via libftdi1 on Windows

 RDM Tests:
 * Fix a bug in the RDM tests when a responder supports sensor PIDs #1388
 * Fix a bug in imports for rdm_responder_test.py

 Bugs:
 * Fix the build on Windows
 * Entire codebase now passes codespell testing
 * Fix a Python 3 bug in the API
 * Fix a few longstanding bugs in the Python ola_rdm_discover
 * Correct the OSC config option name for output format in the documentation
 * Use top_builddir in more places in the build
 * Use libusb_set_option where available
 * Fix the udev rules
 * Downstream Debian fixes
 * Fix an incorrect response if we have the sensor PIDs but no sensors
 * Stop trying to build opendmx kernel plugin on Windows
 * Prefer Avahi over Bonjour for our DiscoveryAgent, in case the Bonjour
   version is actually just Avahi's compatibility layer
 * Clean up the libolaserver pkg-config file

 Internal:
 * Various minor log formatting tidying up

07/01/2018 ola-0.10.6

python/examples/ola_plugin_info.py
#!/usr/bin/env python
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Library General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
#
# ola_plugin_info.py
# Copyright (C) 2005 Simon Newton

"""Lists the loaded plugins."""

from __future__ import print_function

import getopt
import sys
import textwrap

from ola.ClientWrapper import ClientWrapper

__author__ = 'nomis52@gmail.com (Simon Newton)'

wrapper = None


def Usage():
  print(textwrap.dedent("""
  Usage: ola_plugin_info.py [--plugin <plugin_id>]

  Display a list of plugins, or a description for a particular plugin.

  -h, --help                Display this help message and exit.
  -p, --plugin <plugin_id>  Plugin ID number."""))


def Plugins(status, plugins):
  if status.Succeeded():
    for plugin in plugins:
      print('%d %s' % (plugin.id, plugin.name))
  else:
    print('Error: %s' % status.message, file=sys.stderr)

  global wrapper
  if wrapper:
    wrapper.Stop()


def PluginDescription(status, description):
  if status.Succeeded():
    print(description)
  else:
    print('Error: %s' % status.message, file=sys.stderr)

  global wrapper
  if wrapper:
    wrapper.Stop()


def main():
  try:
      opts, args = getopt.getopt(sys.argv[1:], "hp:", ["help", "plugin="])
  except getopt.GetoptError as err:
    print(str(err))
    Usage()
    sys.exit(2)

  plugin = None
  for o, a in opts:
    if o in ("-h", "--help"):
      Usage()
      sys.exit()
    elif o in ("-p", "--plugin"):
      plugin = int(a)

  global wrapper
  wrapper = ClientWrapper()
  client = wrapper.Client()

  if plugin is not None:
    client.PluginDescription(PluginDescription, plugin)
  else:
    client.FetchPlugins(Plugins)

  wrapper.Run()


if __name__ == "__main__":
  main()
